docs/src/transforms.md

```diff
@@ -0,0 +1,246 @@
+# [Transforms](@id about-transforms)
+
+A `Transform` defines a transformation of data for feature engineering purposes.
+Some examples are scaling, periodic functions, linear combination, and one-hot encoding.
+
+```@meta
+DocTestSetup = quote
+    using DataFrames
+    using Dates
+    using FeatureTransforms
+end
+```
+
+## Defining a transform
+
+A `Transform` often has one or more parameters.
+For example, the following defines a squaring operation (i.e. raise to the power of 2):
+
+```julia-repl
+julia> p = Power(2);
+```
+
+## Methods to apply a transform
+
+Given some data `x`, there are three main methods to apply a transform.
+Firstly, it can be applied in a non-mutating fashion using `apply`:
+
+```jldoctest transforms
+julia> p = Power(2);
+
+julia> x = [1.0, 2.0, 3.0];
+
+julia> FeatureTransforms.apply(x, p)
+3-element Array{Float64,1}:
+ 1.0
+ 4.0
+ 9.0
+
+julia> x
+3-element Array{Float64,1}:
+ 1.0
+ 2.0
+ 3.0
+```
+
+Equivalently, the `Transform` object can be called directly on the data:
+
+```jldoctest transforms
+julia> p(x)
+3-element Array{Float64,1}:
+ 1.0
+ 4.0
+ 9.0
+```
+
+Alternatively, the data can be mutated using the `apply!` method.
+
+!!! note
+
+    Not all `Transform` subtypes support mutation.
+
+```jldoctest transforms
+julia> FeatureTransforms.apply!(x, p)
+3-element Array{Float64,1}:
+ 1.0
+ 4.0
+ 9.0
+
+julia> x
+3-element Array{Float64,1}:
+ 1.0
+ 4.0
+ 9.0
+```
+
+A single `Transform` instance can be applied to different data types, with support for `AbstractArray`s and [`Table`s](https://github.com/JuliaData/Tables.jl).
```

Comment:
this reads like it should be the first sentence in the next section?

Reply:
I wanted to lead into the next section with this, but for a user manual it's probably better to put it in the next section.

Reply:
Also it's talking about both arrays and tables, and the next section is just arrays.

Reply:
Leaving this for now.